Buildings/Controls/OBC/Utilities/PIDWithAutotuning/Relay/HalfPeriodRatio.mo

```diff
@@ -0,0 +1,182 @@
+within Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay;
+block HalfPeriodRatio
+  "Calculate the half period ratio of a response from a relay controller"
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput tOn
+    "Length for the On period"
+    annotation (Placement(transformation(extent={{-140,40},{-100,80}}),
+    iconTransformation(extent={{-140,40},{-100,80}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput tOff
+    "Length for the Off period"
+    annotation (Placement(transformation(extent={{-140,-80},{-100,-40}}),
+    iconTransformation(extent={{-140,-80},{-100,-40}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput rho
+    "Real signal of the half period ratio"
+    annotation (Placement(transformation(extent={{100,40},{140,80}}),
+        iconTransformation(extent={{100,50},{120,70}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput triSta
+    "A boolean signal, true if the tuning starts" annotation (Placement(
+        transformation(extent={{100,-20},{140,20}}), iconTransformation(extent=
+            {{100,-10},{120,10}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanOutput triEnd
+    "A boolean signal, true if the tuning completes" annotation (Placement(
+        transformation(extent={{100,-80},{140,-40}}), iconTransformation(extent=
+           {{100,-70},{120,-50}})));
+  Buildings.Controls.OBC.CDL.Continuous.Min tmin
+    "Minimum value of the length for the On period and the length for the off period "
+    annotation (Placement(transformation(extent={{-80,20},{-60,40}})));
+  Buildings.Controls.OBC.CDL.Continuous.Greater gretOntOff
```

Comment:
Does it have the same purpose as instance `gretmaxtOntOff`?

Reply:
No, I modified the comment into "check if both the length for the On period and the length for the off period are larger than 0"